src/Components/Quiz/CurrentQuestion.tsx

```diff
@@ -0,0 +1,157 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router';
+import { useQuiz } from '../../context/quizContext';
+import { Quiz } from '../../database/database.type';
+
+type Prop = {
+	currentQuiz: Quiz;
+};
+
+export const CurrentQuestion = ({ currentQuiz }: Prop) => {
+	const navigate = useNavigate();
+	const { quizState, quizDispatch } = useQuiz();
+	const [disableButtons, setDisableButtons] = useState<boolean>(false);
+	const [optionId, setOptionId] = useState<string>('');
+	const nextQuestion = () => {
+		if (currentQuiz) {
+			quizState.currentQuestionNumber === currentQuiz.questions.length - 1
+				? quizDispatch({ type: 'INCREMENT_QUESTION_NUMBER', payload: -1 })
+				: quizDispatch({ type: 'INCREMENT_QUESTION_NUMBER' });
+		}
+		setDisableButtons(false);
+		if (!optionId) {
+			quizDispatch({
+				type: 'UPDATE_RESULT',
+				payload: {
+					id: currentQuiz.questions[quizState.currentQuestionNumber].id,
+					hasTaken: false,
+					selectedOption: '',
+					correctOption: currentQuiz.questions[
+						quizState.currentQuestionNumber
+					].options.find((option) => option.isRight)?.id,
```

Comment:
Can extract this out in a different function and return the value needed, to make it a little less complex.

Reply:
yes makes sense, will change